### ISSUE TITLE
[Calendar] Support range property

### ISSUE THREAD

**Issue by DanielleRameau** [sl-design-system/components#3396]
Support range property in calendar. See design: https://www.figma.com/design/CHpKrPIdXdbV2u7X8vizKI/Components-2.0?node-id=10364-251296&t=YDlFGMgaopcFYpPx-4

- [ ] Single selection of one day should be available as well
- [ ] Calendar should have a `mode="range"` property 
- [ ] Calendar should have a `range` property that is an array of 2 dates (order of the dates shouldn't matter)
- [ ] When in range mode after the first click, hovering over another day should indicate what the range will be when you click a second time
- [ ] Escape should abort the range selection after the first click

**Examples**
https://mui.com/x/react-date-pickers/date-range-picker/
https://dna.iddinkgroup.com/?path=/story/components-calendar--range

**Comment by a11ymiko** [comment]
Accessibility features that Date Range should have:

- [ ] all the a11y features we already have for Calendar

- [ ] when both start and end dates are selected `aria-describedby` should be added to start and end date buttons. Text of `aria-describedby` should be: 'Start of range' and 'End of range'. Other dates in. range don't need to have `aria-describedby` added to them

- [ ]  when first date is selected `sl-announcer` should announce: 'DATE selected. Select second date.'. When the second date is selected `sl-announcer` should announce 'You selected range from START DATE to END DATE'.

- [ ] focus order: With `tab` key user should be able to go to first active date of each of the calendars, so if there are three calendars visible the first `tab` press is setting focus on first active element of first calendar, next `tab` press sets focus on the begining of second calendar, and the third `tab` press sets focus on the first element of the third calendar.

- [ ] keyboard operability: user should be able to navigate calnedars with arrows. Using left and right arrows should change focus to the previous/next day and when this day is in other calendar than focus should move to that calendar. Using up and down arrows should change focus to previous/next week. If previouse/next week day is in other calendar then focus should be moved to that calendar.
